### PR TITLE
Add CI

### PR DESCRIPTION
**Pull request by vincentmacri**
This commit adds the default GitHub actions for Django and Python apps, with minor tweaks for versions.

### CODE REVIEW

File: .github/workflows/django.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install Dependencies
+      run: |
+        python -m pip install --upgrade pip
```
Comment (tyxchen) [inline .github/workflows/django.yml:38]: Do we always want to be using the newest version of pip? Should we fix this at a specific pip version (e.g. 20.0.2)?

File: .github/workflows/pythonapp.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
```
Comment (tyxchen) [inline .github/workflows/pythonapp.yml]: I don't quite understand what the purpose of this file is, seeing that all our tests are going to be run through django. I do like having flake8 check our code style though.

File: README.md [hunk 3]
```diff
@@ -2,7 +2,9 @@
 The backend portion of `slugline`, math**NEWS**'s best and only Django/React based publishing website.
 
 ## Running
-- Create a `virtualenv` by running `virtualenv <environment location>` and run `<environment location>/scripts/activate` to activate it. Python 3.7+ is recommmended.
+- Create a `virtualenv` by running `virtualenv <environment location>`. Python 3.7+ is recommmended. Depending on your OS run the following to activate the virtualenv:
+	- Windows: `<environment location>/scripts/activate`
+	- Linux: `source <environment location>/bin/activate`
 - Run `pip install -r requirements.txt` to download depdendencies.
```
Comment (ktrieu) [inline README.md]: While we're changing the README, I spelt dependencies wrong, apparently.

File: .github/workflows/pythonapp.yml (hunk 2 above)
Comment (ktrieu) [reply .github/workflows/pythonapp.yml]: If we're doing flake8, may as well run black as well.

**Comment by tyxchen** [comment]
By the way, is it possible to add a cache here as well?

File: .github/workflows/django.yml (hunk 1 above)
Comment (vincentmacri) [reply .github/workflows/django.yml:38]: Pip is the package manager, so I don't see why we wouldn't always want it updated.

File: .github/workflows/pythonapp.yml (hunk 2 above)
Comment (vincentmacri) [reply .github/workflows/pythonapp.yml]: @tyxchen the purpose is this was the default Python CI GitHub Actions script. I've removed the pytest part, but kept the code style checks.

**Comment by vincentmacri** [comment]
I think this is ready to merge, fixing the one failing test requires running auto formatting on a bunch of files and I'd prefer to have that be a separate PR.

File: .github/workflows/flake.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    - name: Display Python version
+      run: python -c "import sys; print(sys.version)"
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+
+    - name: Lint with flake8
```
Comment (ktrieu) [inline .github/workflows/flake.yml:44]: Are we not going to do `flake8` with the pre commit hook as well?

File: .github/workflows/black.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        restore-keys: |
+          ${{ runner.os }}-black-${{ matrix.python-version }}-pip-
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Display Python version
+      run: python -c "import sys; print(sys.version)"
```
Comment (tyxchen) [inline .github/workflows/black.yml]: Would recommend `python -c "import platform; print(platform.python_version(), platform.platform())`

File: .github/workflows/django.yml (hunk 1 above)
Comment (tyxchen) [reply .github/workflows/django.yml:38]: For stability, I suppose --- though in install instructions we could always include `pip install --upgrade pip`.

**Comment by tyxchen** [comment]
Ooh, I just realized black formats migrations. I don't know if I like that. Any way to exclude black from formatting a specific directory?

File: .github/workflows/flake.yml (hunk 4 above)
Comment (vincentmacri) [reply .github/workflows/flake.yml:44]: I think we only need one formatter to run on commits.

File: .github/workflows/black.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    branches: [ '**' ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-16.04
+    strategy:
+      max-parallel: 4
+      matrix:
+        python-version: [3.8]
```
Comment (tyxchen) [inline .github/workflows/black.yml]: Why does the Python version differ for this workflow from the others?

Comment (vincentmacri) [reply .github/workflows/black.yml]: I thought black would complain if I ran it on multiple versions, I'll see if I can run it on multiple versions though.

Comment (vincentmacri) [reply .github/workflows/black.yml]: It complains, and also seems unnecessary, since it's just a code style check.

Comment (tyxchen) [reply .github/workflows/black.yml]: The complaining looks to be due to the python version in the pre-commit hook being fixed on 3.8, which is something we should probably fix. I agree on the unnecessary part though --- we can probably just run it on 3.x and be done with it.

**Review by tyxchen — APPROVED** [review]

**Review by ktrieu — APPROVED** [review]

**Pull request merged**